### PR TITLE
Split sui system state file

### PR DESCRIPTION
**Pull request by lxfind**
A few more refactoring is coming.
Before that, first split the growing sui system state file into a dedicated sub module with multiple files:
1. Put a few common Move types such as Table into collection_types.rs
2. Put all the V1 data structures in a dedicated V1 file.

### CODE REVIEW

**Review by bmwill — APPROVED** [review]

**Comment by bmwill** [comment]
>Before that, first split the growing sui system state file into a dedicated sub crate with multiple files:

Slight nit: You're moving it into a sub module not another crate

File: crates/sui-types/src/collection_types.rs [hunk 1]
```diff
@@ -22,3 +24,68 @@ pub struct Entry<K, V> {
 pub struct VecSet<T> {
     contents: Vec<T>,
 }
+
+/// Rust version of the Move std::option::Option type.
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq, JsonSchema)]
+pub struct MoveOption<T> {
```
Comment (emmazzz) [inline crates/sui-types/src/collection_types.rs:30]: Not sure if Option is considered a collection type but I'm fine with it staying here.

File: crates/sui-types/src/sui_system_state/mod.rs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+pub const INIT_SYSTEM_STATE_VERSION: u64 = 1;
+
+/// Rust version of the Move sui::sui_system::SuiSystemState type
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct SuiSystemStateWrapper {
+    pub id: UID,
+    pub version: u64,
+}
+
+impl SuiSystemStateWrapper {
```
Comment (emmazzz) [inline crates/sui-types/src/sui_system_state/mod.rs:44]: I find it slightly confusing the relationship between `SuiSystemStateWrapper`, `SuiSystemState` enum and `SuiSystemStateTrait`, and when each should be used. Can we have a few lines of comments explaining it?

File: crates/sui-types/src/collection_types.rs (hunk 1 above)
Comment (lxfind) [reply crates/sui-types/src/collection_types.rs:30]: Yeah maybe we will eventually add another file called common_move_types?

**Pull request merged**